### PR TITLE
Fix Distributed Endoscopy Tool Tracking Test Failures

### PR DESCRIPTION
**Pull request by mocsharp**
This PR implements two options to provide input data and config file and fixes the test failures:

- Allows users to provide input data and config file via 
  - arguments: `--data` and `--config`, or
  - environment variables: `HOLOSCAN_INPUT_PATH` and `HOLOSCAN_CONFIG_PATH` (as prescribed by [HAP specifications](https://docs.nvidia.com/holoscan/sdk-user-guide/cli/hap.html))
  - fix test cases and add additional ones to cover both use cases
  - remove existing frame validation coverage since this app does not support recording

### CODE REVIEW

File: applications/endoscopy_tool_tracking_distributed/cpp/CMakeLists.txt [hunk 1]
```diff
@@ -58,43 +58,21 @@ add_dependencies(endoscopy_tool_tracking_distributed endoscopy_tool_tracking_dis
...
-      --validation_frames_dir ${VALIDATION_FRAMES_DIR}
-      WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
-  )
-
-  set_tests_properties(endoscopy_tool_tracking_distributed_cpp_render_test PROPERTIES
-  DEPENDS endoscopy_tool_tracking_distributed_cpp_test
-  PASS_REGULAR_EXPRESSION "Valid video output!"
-  )
-
 endif()
```
Comment (gigony) [inline applications/endoscopy_tool_tracking_distributed/cpp/CMakeLists.txt:78]: nit: need a blank line at the end.

File: applications/endoscopy_tool_tracking_distributed/cpp/metadata.json [hunk 2]
```diff
@@ -32,7 +32,7 @@
 			]
 		},
 		"run": {
-			"command": "HOLOSCAN_INPUT_PATH=<holohub_data_dir>/endoscopy <holohub_app_bin>/endoscopy_tool_tracking_distributed",
+			"command": "<holohub_app_bin>/endoscopy_tool_tracking_distributed",
```
Comment (gigony) [inline applications/endoscopy_tool_tracking_distributed/cpp/metadata.json:35]: Does the user need to set the environment variable (`HOLOSCAN_INPUT_PATH`) before executing this command on the host system (without a Docker container)? How will the user know about the requirement for the environment variable if it is not specified in this command?

**Review by gigony — APPROVED** [review]
Thanks @mocsharp !
Looks good to me with a question on the example usage.

File: applications/endoscopy_tool_tracking_distributed/cpp/metadata.json (hunk 2 above)
Comment (jjomier) [reply applications/endoscopy_tool_tracking_distributed/cpp/metadata.json:35]: We should add/use the `--data` from the command line. This is what we have for most of the applications.
See https://github.com/nvidia-holoscan/holohub/blob/main/applications/endoscopy_tool_tracking/cpp/metadata.json#L35 for reference

Comment (mocsharp) [reply applications/endoscopy_tool_tracking_distributed/cpp/metadata.json:35]: Okay. Gigon confirms that arguments still work; here's an excerpt from the `application.hpp` source code:

>    * This constructor parses the command line for flags that are recognized by App Driver/Worker,
>   * and removes all recognized flags so users can use the remaining flags for their own purposes.

Let me update the app to use `--data`

Comment (jjomier) [reply applications/endoscopy_tool_tracking_distributed/cpp/metadata.json:35]: Awesome - thanks for looking into this!

**Review by gigony — APPROVED** [review]
It looks great to me!
Thank you @mocsharp for fixing the issue!

File: applications/endoscopy_tool_tracking_distributed/cpp/CMakeLists.txt [hunk 3]
```diff
@@ -58,43 +58,29 @@ add_dependencies(endoscopy_tool_tracking_distributed endoscopy_tool_tracking_dis
...
            WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
-  set_tests_properties(endoscopy_tool_tracking_distributed_cpp_test PROPERTIES
+
+  set_tests_properties(endoscopy_tool_tracking_distributed_cpp_test_args PROPERTIES
                        PASS_REGULAR_EXPRESSION "Reach end of file or playback count reaches to the limit. Stop ticking."
                        FAIL_REGULAR_EXPRESSION "[^a-z]Error;ERROR;Failed")
 
-  # Add a test to check the validity of the frames
-  add_test(NAME endoscopy_tool_tracking_distributed_cpp_render_test
-      COMMAND python3 ${CMAKE_SOURCE_DIR}/utilities/video_validation.py
```
Comment (tbirdso) [inline applications/endoscopy_tool_tracking_distributed/cpp/CMakeLists.txt:87]: Same question, why remove frame tests?

**Review by tbirdso — COMMENTED** [review]
This looks great! A couple of questions about test coverage.

File: applications/endoscopy_tool_tracking_distributed/python/CMakeLists.txt [hunk 4]
```diff
@@ -18,48 +18,33 @@ if(BUILD_TESTING)
...
-  set_tests_properties(endoscopy_tool_tracking_distributed_python_test PROPERTIES
-                PASS_REGULAR_EXPRESSION "Reach end of file or playback count reaches to the limit. Stop ticking.;"
-                FAIL_REGULAR_EXPRESSION "[^a-z]Error;ERROR;Failed")
+  add_test(NAME endoscopy_tool_tracking_distributed_python_test_env_vars
+          COMMAND python3 ${CMAKE_CURRENT_SOURCE_DIR}/endoscopy_tool_tracking.py --data "${HOLOHUB_DATA_DIR}/endoscopy" --config "${CMAKE_CURRENT_BINARY_DIR}/endoscopy_tool_tracking_testing.yaml"
+          WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
 
-  # Add a test to check the validity of the frames
-  add_test(NAME endoscopy_tool_tracking_distributed_python_render_test
-    COMMAND python3 ${CMAKE_SOURCE_DIR}/utilities/video_validation.py
```
[the remark this answers is not in the document]
Comment (mocsharp) [reply applications/endoscopy_tool_tracking_distributed/python/CMakeLists.txt:52]: Yes, this app does not support recording like the non-distributed version of the endoscopy tool tracking app.

File: applications/endoscopy_tool_tracking_distributed/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+ARG BASE_IMAGE
+ARG GPU_TYPE
+
+FROM ${BASE_IMAGE} AS base
+
+ARG DEBIAN_FRONTEND=noninteractive
+
+# Holohub Environment Variables
+ENV HOLOSCAN_INPUT_PATH=/workspace/holohub/data/endoscopy
```
[the remark this answers is not in the document]
Comment (mocsharp) [reply applications/endoscopy_tool_tracking_distributed/Dockerfile:31]: I have updated the Python test with `HOLOSCAN_INPUT_PATH` and remove the Dockerfile. 

I kept it thinking to have a way to demo the use of HAP.

**Pull request merged**